scrapy/commands/fetch.py

```diff
@@ -47,8 +48,15 @@ def _print_bytes(self, bytes_):
         sys.stdout.buffer.write(bytes_ + b'\n')
 
     def run(self, args, opts):
-        if len(args) != 1 or not is_url(args[0]):
+        if len(args) != 1:
             raise UsageError()
+
+        settings = get_project_settings()
+        supported_protocols = settings.getdict('DOWNLOAD_HANDLERS_BASE').keys()
+        supported_protocols |= settings.getdict('DOWNLOAD_HANDLERS').keys()
```

Comment:
See `settings.getwithbase`. Using that, extracting this code into a function is less important, as you could simplify the line below as:

```python
        if not is_uri(args[0], settings.getwithbase('DOWNLOAD_HANDLERS')):
```